contracts/Challenge/PLCRVotingChallenge.sol

```diff
@@ -0,0 +1,204 @@
+
+    // ============
+    // DATA STRUCTURES:
+    // ============
+
+    using AttributeStore for AttributeStore.Data;
+    using DLL for DLL.Data;
+    using SafeMath for uint;
+
+    // ============
+    // GLOBAL VARIABLES:
+    // ============
+
+    address public challenger;     /// the address of the challenger
+    address public listingOwner;   /// the address of the listingOwner
+    Registry public registry;
+    PLCRVoting public voting;      /// address of PLCRVoting Contract
+    uint public pollID;            /// pollID of PLCRVoting
+    bool challengeResolved;        /// true is challenge has officially been resolved to passed or failed
+    uint public commitEndDate;     /// expiration date of commit period for poll
+    uint public revealEndDate;     /// expiration date of reveal period for poll
+    uint public voteQuorum;	    /// number of votes required for a proposal to pass
+    uint public rewardPool;        /// pool of tokens to be distributed to winning voters
+    uint public challengerStake;   /// number of tokens at stake for either party during challenge
+    uint public votesFor;		    /// tally of votes supporting proposal
+    uint public votesAgainst;      /// tally of votes countering proposal
+
+    uint public voterTokensClaimed;
+    uint public voterRewardsClaimed;
+
+    uint public commitStageLen;
+    uint public revealStageLen;
+
+    mapping(address => bool) public didCommit;     /// indicates whether an address committed a vote for this poll
+    mapping(address => bool) public didReveal;     /// indicates whether an address revealed a vote for this poll
+
+    mapping(address => uint) public voteTokenBalance; // maps user's address to voteToken balance
+    mapping(address => bool) public tokenClaims;   // Indicates whether a voter has claimed a reward yet
+
+    AttributeStore.Data store;
+
+    // ============
+    // MODIFIERS:
+    // ============
+
+    modifier onlyChallenger() {
+        require(msg.sender == challenger);
+        _;
+    }
+
+    // ============
+    // CONSTRUCTOR:
+    // ============
+
+    /**
+    @dev Initializes voteQuorum, commitDuration, revealDuration, and pollNonce in addition to token contract and trusted mapping
+    */
+    function PLCRVotingChallenge(address _challenger, address _listingOwner, Registry _registry, Parameterizer _parameterizer) public {
+        challenger = _challenger;
+        listingOwner = _listingOwner;
+        registry = _registry;
+        voting = _parameterizer.voting();
+
+        commitStageLen  = _parameterizer.get("commitStageLen");
+        revealStageLen  = _parameterizer.get("revealStageLen");
+        voteQuorum      = _parameterizer.get("voteQuorum");
+        challengerStake = _parameterizer.get("minDeposit");
+        pollID          = voting.startPoll( voteQuorum, commitStageLen, revealStageLen);
+
+        uint oneHundred = 100; // Kludge that we need to use SafeMath
```

Comment:
what?

Reply:
lol that's a copy/paste from their code .. just moved to a different file